pages/Developer-Guide-(Compiling).md

```diff
@@ -0,0 +1,56 @@
+This is a part of our Developer Guide, you can find a full overview on our [main page](https://github.com/Slimefun/Slimefun4/wiki/Developer-Guide).
+
+This guide assumes you are using Maven for your project. 
+
+## Compiling
+
+In order to test your addon, you need to compile the addon into a `.jar` file (the files that servers recognize as plugins).<br>
+If you chose to use Maven for your project (as we suggested in [Part 1](https://github.com/Slimefun/Wiki/blob/master/pages/Developer-Guide-(1-Project-Setup).md)), you can simply run the following command:
+
+```
+$ mvn clean package
+```
+Your compiled `.jar` can be found in the `target/` directory in the project folder. 
+
+IntelliJ also provides a way to compile your plugin.<br>
+On the right side of the editor window, click the button that says Maven. (If there is no such button, go to the top bar and navigate to `View > Tool Windows > Maven`)<br>
+Now in the project folder, open the folder called `Lifecycle` and double click `clean`. Once that is complete, double click `package`.
+
+Once again, your compiled `.jar` can be found in the `target/` directory in the project folder. 
+
+## Creating a testing environment
+The next step is to run a server with your freshly compiled addon.<br>
+We recommend [Paper](https://papermc.io/downloads) as your server software. <br>
+Do note that only Spigot and its forks (Paper, Purpur, etc.) are supported by Slimefun and its addons.
+
+The server should be in the form of a `.jar` file.<br>
+Create a new folder somewhere dedicated for your server, and place the `.jar` file there.
```

Comment:
```suggestion
Create a new folder somewhere dedicated for your server, and place the downloaded server `.jar` file there.
```

to avoid confusion between their .jar and the server .jar
(Maybe go as far as to set up a server first before compiling the addon? - open question)